client/verta/tests/unit_tests/registry/test_registered_model_version.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+
+""" Unit tests for the RegisteredModelVersion class. """
+
+from string import ascii_letters
+from typing import List
+from unittest.mock import patch
+
+import hypothesis
+import hypothesis.strategies as st
+
+from verta._internal_utils._utils import timestamp_to_str
+from verta._protos.public.registry import RegistryService_pb2
+from verta.registry.entities import RegisteredModelVersion
+
+from ..strategies import (
+    artifact_protos,
+    code_blob_proto,
+    int64,
+    model_artifact_proto,
+    uint64,
+)
+
+
+@st.composite
+def model_ver_proto(draw) -> RegistryService_pb2.ModelVersion:
+    """Generate a mocked ModelVersion protobuf object."""
+    return RegistryService_pb2.ModelVersion(
+        id=draw(uint64()),
+        registered_model_id=draw(uint64()),
+        version=draw(st.text(ascii_letters)),
+
+        time_updated=draw(int64()),
+        time_created=draw(int64()),
+
+        labels=draw(st.lists(st.text(ascii_letters), unique=True)),
+
+        input_description=draw(st.text(ascii_letters)),
+        output_description=draw(st.text(ascii_letters)),
+        hide_input_label=draw(st.booleans()),
+        hide_output_label=draw(st.booleans()),
+
+        model=draw(st.one_of(st.none(), model_artifact_proto())),
+        artifacts=draw(artifact_protos()),
+        datasets=draw(artifact_protos()),
+        code_blob_map=draw(st.dictionaries(st.text(ascii_letters), code_blob_proto())),
+    )
+
+
+@patch.object(RegisteredModelVersion, "_refresh_cache", return_value=None)
+@hypothesis.given(model_ver_proto=model_ver_proto(), workspace=st.text(ascii_letters, min_size=1))
+def test_repr(mock_conn, mock_config, model_ver_proto, workspace):
+    """
+    Verify that RegisteredModelVersion.__repr__() renders expected fields and correct values.
+
+    This test does not yet cover all available fields, but exists in its current form to cover newly-added model catalog fields.
+
+    """
+    model_ver = RegisteredModelVersion(conn=mock_conn, conf=mock_config, msg=model_ver_proto)
```

Comment:
This ought to pave the way for fully-mocked `RegisteredModelVersion` tests.

Reply:
This is really cool.  I've been digging into `hypothesis` documentation and it seems like a rather brilliant testing solution.  I've written extra fixtures to mock edge cases and used them to parameterize queries but this approach is like that only a thousand times more comprehensive.